x-pack/plugins/apm/ftr_e2e/cypress/integration/read_only_user/service_overview/service_overview.spec.ts

```diff
@@ -57,4 +57,23 @@ describe('Service Overview', () => {
       'Worker'
     );
   });
+
+  it('hides dependency tab when RUM service', () => {
+    cy.intercept('GET', '/api/apm/services/opbeans-rum/agent').as(
+      'agentRequest'
+    );
+    cy.visit(
+      url.format({
+        pathname: '/app/apm/services/opbeans-rum/overview',
+        query: { rangeFrom: start, rangeTo: end },
+      })
+    );
+    cy.contains('Overview');
+    cy.contains('Transactions');
+    cy.contains('Error');
+    cy.contains('Service Map');
+    // Waits until the agent request is finished to check the tab.
+    cy.wait('@agentRequest');
```

Comment:
This syntax bothers - but that's just how Cypress is I guess. If I had designed this I would have done something like:

```ts
const endpoint = cy.intercept('GET', '/api/apm/services/opbeans-rum/agent')
// ...
cy.wait(endpoint);
```

To avoid relying on strings that don't have any type-safety.

Reply:
> This syntax bothers - but that's just how Cypress is I guess. If I had designed this I would have done something like:
> 
> 
> 
> ```ts
> 
> const agentEndpoint = cy.intercept('GET', '/api/apm/services/opbeans-rum/agent')
> 
> // ...
> 
> cy.wait(agentEndpoint);
> 
> ```
> 
> 
> 
> To avoid relying on strings that don't have any type-safety.
> 
> 
> 
> 

Yes, I agree with you, relying on strings aren't the best

Reply:
Might as well raise it as an enhancement request :)
https://github.com/cypress-io/cypress/issues/17866